packages/social-previews/src/search-preview/index.jsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { useTranslate } from 'i18n-calypso';
+import React from 'react';
+import { firstValid, hardTruncation, shortEnough, truncatedAtSpace } from '../helpers';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const TITLE_LENGTH = 63;
+const SNIPPET_LENGTH = 160;
+
+const googleTitle = firstValid(
+	shortEnough( TITLE_LENGTH ),
+	truncatedAtSpace( TITLE_LENGTH - 40, TITLE_LENGTH + 10 ),
+	hardTruncation( TITLE_LENGTH )
+);
+
+const googleSnippet = firstValid(
+	shortEnough( SNIPPET_LENGTH ),
+	truncatedAtSpace( SNIPPET_LENGTH - 80, SNIPPET_LENGTH + 10 ),
+	hardTruncation( SNIPPET_LENGTH )
+);
+
+const googleUrl = hardTruncation( 79 );
+
+export default function SearchPreview( { snippet, title, url, description } ) {
+	const translate = useTranslate();
+
+	// Enable description prop to provide consistency with other components within the
+	// package which use `description`.
+	if ( description ) {
+		snippet = description;
+	}
+
+	return (
+		<div className="search-preview">
+			<h2 className="search-preview__header">{ translate( 'Search Preview' ) }</h2>
+			<div className="search-preview__display">
+				<div className="search-preview__title">{ googleTitle( title ) }</div>
+				<div className="search-preview__url">{ googleUrl( url ) } ▾</div>
+				<div className="search-preview__snippet">{ googleSnippet( snippet || '' ) }</div>
```

Comment:
It seems that `description` is more generic than `snippet` and we will deprecate the `snippet` var. In that case, I think it is preferable to do

`export default function SearchPreview( { snippet, description = snippet, title, url } ) {`

and remove

```
if ( snippet ) {
		description = snippet;
	}
```

and change it both in the class and the content 
`<div className="search-preview__description">{ googleSnippet( description || '' ) }</div>`

We should also document clearly that `snippet` will be deprecated by `description`.

That way, we can easily remove the `snippet` references in the future

Reply:
I toyed with both ideas. The key is we have a consistent API that allows the use of `description`.

To keep the change set in this PR _minimal_ I'd suggest we leave "as is" and then raise a follow up PR which deprecates `snippet` and updates everything to use `description` as the first class API.

If you're up for that would you like to raise an Issue? We may already have a `deprecate` package somewhere. I think in `@wordpress` packages?

Reply:
That sounds good! Isn't

`export default function SearchPreview( { snippet, description = snippet, title, url } ) {`

equivalent to 

```
if ( snippet ) {
		description = snippet;
	}
```

Reply:
Actually I pushed a PR addressing that change #44309. Can you review it so I can merge / deploy and you can rebase?

Reply:
I have merged / deployed #44309 so you can rebase now! And remove the `snippet` variable from your npm package too!

Reply:
I've pulled and updated the components. Want to check I caught everything?